pkg/guardian/metrics.go

```diff
@@ -58,6 +71,18 @@ type DataDogReporter struct {
 	c           chan func()
 }
 
+type MetricOption struct {
+	additionalTags []string
+}
+
+type MetricOptionSetter func(mo *MetricOption)
+
+func WithRemoteAddress(remoteAddress string) MetricOptionSetter {
```

Comment:
This is probably a very high cardinality tag.

Reply:
Yeah, I guess I thought that was a tradeoff that was worth exploring since it could be helpful to have these metrics as we rollout this new feature. I'll reconsider this.